File: lib/controller.py
from evdev import InputDevice, ecodes

DEADZONE = 10

def main():
    try:
        device = InputDevice('/dev/input/event25')
        print(f"Using device: {device.name}")

        for event in device.read_loop():
            if event.type == ecodes.EV_KEY or event.type == ecodes.EV_ABS:
                if event.code in [ecodes.ABS_X, ecodes.ABS_Y, ecodes.ABS_Z, ecodes.ABS_RX]:
                    if event.value > (127 + DEADZONE) or event.value < (127 - DEADZONE):
                        print(f"{event.code}, {event.value}")
                elif event.code in [ecodes.ABS_WHEEL, ecodes.ABS_RZ]:
                    print(f"{event.code}, {event.value}")
                else:
                    print(f"{event.code}, {event.value}")

    except FileNotFoundError:
        print("E Controller not found")
        return 1

    return 0

if __name__ == "__main__":
    main()

File: src/controller.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>
#include <linux/input.h>

int DEADZONE = 10;

int main()
{
    int fd = open("/dev/input/event25", O_RDONLY);
    if (fd < 0)
    {
        printf("E Controller not found\n");
        return 1;
    }

    while (1)
    {
        struct input_event ev;
        ssize_t bytesRead = read(fd, &ev, sizeof(struct input_event));
        if (bytesRead < sizeof(struct input_event))
        {
            printf("E Failed to read inputs\n");
            break;
        }

        if (ev.type == EV_KEY || ev.type == EV_ABS)
        {   if (ev.code == 0 || ev.code == 1 || ev.code == 3 || ev.code ==4)
            {
                if (ev.value > (127 + DEADZONE) || ev.value < (127 - DEADZONE))
                {
                    printf("%d, %d\n", ev.code, ev.value);
                }
            }
            else if (ev.code == 2 || ev.code == 5)
            {
                printf("%d, %d\n", ev.code, ev.value);
            }
            else
            {
                printf("%d, %d\n", ev.code, ev.value);
            }
        }
    }
    close(fd);
    return 0;
}